### ISSUE TITLE
fetchmail fails with python3

### ISSUE THREAD

**Issue by linkit-odoo** [odoo/odoo#20857]
Impacted versions: 11.0

Steps to reproduce: fetchmal fails consistently while downloading messages

Current behavior: TypeError

Proposed patch for file addons/fetchmail/models/fetchmail.py:
```
--- fetchmail.py.orig	2017-10-30 20:53:32.590000000 +0000
+++ fetchmail.py	2017-10-30 20:53:53.890000000 +0000
@@ -204,7 +204,7 @@
                         pop_server.list()
                         for num in range(1, min(MAX_POP_MESSAGES, num_messages) + 1):
                             (header, messages, octets) = pop_server.retr(num)
-                            message = '\n'.join(messages)
+                            message = b'\n'.join(messages)
                             res_id = None
                             try:
                                 res_id = MailThread.with_context(**additionnal_context).message_process(server.object_id.model, message, save_original=server.original, strip_attachments=(not server.attach))
```

**Comment by linkit-odoo** [comment]
Further problems arise with fetchmail, even when patched as above.
Messages sent to an help desk POP box get created but only preserving the original subject, losing all body text.

We get constantly this type of error in logs:
```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/addons/fetchmail/models/fetchmail.py", line 210, in fetch_mail
    res_id = MailThread.with_context(**additionnal_context).message_process(server.object_id.model, message, save_original=server.original, strip_attachments=(not server.attach))
  File "/usr/lib/python3/dist-packages/odoo/addons/mail/models/mail_thread.py", line 1295, in message_process
    thread_id = self.message_route_process(msg_txt, msg, routes)
  File "/usr/lib/python3/dist-packages/odoo/addons/mail/models/mail_thread.py", line 1205, in message_route_process
    thread = MessageModel.message_new(message_dict, custom_values)
  File "/opt/odoo/extra-addons/enterprise/helpdesk/models/helpdesk_ticket.py", line 352, in message_new
    ticket = super(HelpdeskTicket, self).message_new(msg, custom_values=values)
  File "/usr/lib/python3/dist-packages/odoo/addons/mail/models/mail_thread.py", line 1331, in message_new
    return RecordModel.create(data)
  File "/opt/odoo/extra-addons/custom/sintetica/models/helpdesk.py", line 80, in create
    ticket = self._create_task_if_not_exists(values, ticket)
  File "/opt/odoo/extra-addons/custom/sintetica/models/helpdesk.py", line 23, in _create_task_if_not_exists
    'date_deadline': values['task_date_deadline'],
KeyError: 'task_date_deadline'
```

**Comment by Yenthe666** [comment]
@xmo-odoo this might be for you.

**Comment by adrianriedo** [comment]
I can confirm the problem
With Odoo 10, the POP works fine and all emails are fetched
With Odoo 11 and the exact same POP configuration, no emails are fetched and we get

2017-11-10 22:20:54,572 1124 INFO rnx odoo.addons.fetchmail.models.fetchmail: General failure when trying to fetch mail from pop server <my-POP-server>.
Traceback (most recent call last):
  File "/opt/odoo/venv/lib/python3.5/site-packages/odoo/addons/fetchmail/models/fetchmail.py", line 207, in fetch_mail
    message = '\n'.join(messages)

In Odoo UI you get no error, only in the logfile we can see that something is not working

**Comment by robertallanroker** [comment]
I have the identical problem.  What is the issue?

**Comment by Yenthe666** [comment]
@robertallanroker the exact issue is coming from not correctly parsing. Fix at https://github.com/odoo-dev/odoo/commit/dde554726b301e42aaa297f899e6fcb83d9e9e63

**Comment by mattolden** [comment]
Had the same issue, changed line 207 as indicated in the fix at odoo-dev/odoo@dde5547 posted by Yenthe666

Retested with 'fetch now' and again only the log shows any error but it is now different than before.  See new error below, my emphasis.

  File "/odoo/odoo-server/addons/fetchmail/models/fetchmail.py", line 207, in fetch_mail
    **(header, messages, octets) = pop_server.retr(num)**

**Comment by Yenthe666** [comment]
@nim-odoo this is relevant for your fix.
